src/components/attendee/MiniProfile/components/MiniProfileModal.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback } from "react";
+import { useHistory } from "react-router-dom";
+import { Card } from "components/attendee/Card";
+import { SpentTime } from "components/shared/SpentTime";
+import firebase from "firebase/compat/app";
+
+import { UserId, UserWithId } from "types/id";
+
+import { generateAttendeeSpaceLandingUrl } from "utils/url";
+
+import { useChatSidebarControls } from "hooks/chats/util/useChatSidebarControls";
+import { useSpaceParams } from "hooks/spaces/useSpaceParams";
+import { useIsCurrentUser } from "hooks/useIsCurrentUser";
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+
+import { UserAvatar } from "components/atoms/UserAvatar";
+
+import { MiniProfileModalLinks } from "./MiniProfileModalLinks";
+
+import styles from "./MiniProfileModal.module.scss";
+
+interface MiniProfileModalProps {
+  profile: UserWithId;
+  userId: UserId;
+}
+
+export const MiniProfileModal: React.FC<MiniProfileModalProps> = ({
+  profile,
+}) => {
+  const { worldSlug, spaceSlug } = useSpaceParams();
+
+  const history = useHistory();
+
+  const { selectRecipientChat } = useChatSidebarControls();
+
+  const { closeUserProfileModal } = useProfileModalControls();
+
+  const openChat = useCallback(() => {
+    selectRecipientChat(profile);
+    closeUserProfileModal();
+  }, [selectRecipientChat, profile, closeUserProfileModal]);
+
+  const logout = useCallback(async () => {
+    await firebase.auth().signOut();
+    closeUserProfileModal();
+
+    history.push(generateAttendeeSpaceLandingUrl(worldSlug, spaceSlug));
+  }, [closeUserProfileModal, history, worldSlug, spaceSlug]);
+
+  const isCurrentUser = useIsCurrentUser(profile.id);
+
+  const actionButton = isCurrentUser ? (
+    <Card.Button variant="danger" onClick={logout}>
+      Log Out
+    </Card.Button>
+  ) : (
+    <Card.Button variant="intensive" onClick={openChat}>
+      Send message
+    </Card.Button>
+  );
```

Comment:
Most of this code is the same, it would be easy to follow it if it's not put into a variable. The `onClick` and inside text can use the `? :` treatment.

Reply:
I wouldn't say it would look better. As instead of 1 ternary operator, there would be 3 of them. But if you don't like the usage of `actionButton` const, I moved the ternary operator directly to the HTML part. Hopefully, it'll look more readable.
https://github.com/sparkletown/sparkle/pull/2953/commits/a52527d7fce74b5f15447723260cc7715c8614b3